source/appController/AMDSAppController.cpp

```diff
@@ -0,0 +1,27 @@
+#include "AMDSAppController.h"
+
+#include "util/AMErrorMonitor.h"
+
+AMDSAppController *AMDSAppController::appController_ = 0;
+
+AMDSAppController *AMDSAppController::appController()
+{
+	if (!appController_)
+		AMErrorMon::error(0, 0, "The app controller is NOT instanced!");
```

Comment:
This is a super class, the instance should be initialized by AMDSClientAppController / AMDSMainAppController. It works like AMBeamline